src/protect/pipeline/ProTECT.py

```diff
@@ -497,11 +498,24 @@ def launch_protect(job, patient_data, univ_options, tool_options):
                          cores=1, disk='100M').encapsulate()
     bam_files['tumor_rna'].addChild(rsem)
     # Define the fusion calling node
-    fusions = job.wrapJobFn(run_fusion_caller, bam_files['tumor_rna'].rv(), univ_options,
-                            'fusion_options', disk='100M', memory='100M', cores=1)
+
+    tool_options['star_fusion']['index'] = tool_options['star']['index']
+    tool_options['fusion_inspector']['index'] = tool_options['star']['index']
+    fusions = job.wrapJobFn(run_star_fusion,
+                            cutadapt.rv(),
+                            bam_files['tumor_rna'].rv('rnaChimeric.out.junction'),
+                            univ_options,
+                            tool_options['star_fusion'],
+                            tool_options['fusion_inspector'],
+                            disk='100M', memory='100M', cores=1)
+
+    fusions.addChild(fastq_deletion_1)
+    fusions.addChild(fastq_deletion_2)
     bam_files['tumor_rna'].addChild(fusions)
     # Define the bam deletion node
-    delete_bam_files['tumor_rna'] = job.wrapJobFn(delete_bams, bam_files['tumor_rna'].rv(),
+    bam_dict = {'rnaAligned.out.bam': bam_files['tumor_rna'].rv('rnaAligned.out.bam')}
+    delete_bam_files['tumor_rna'] = job.wrapJobFn(delete_bams,
+                                                  bam_dict,
                                                   univ_options['patient'], disk='100M',
                                                   memory='100M')
     bam_files['tumor_rna'].addChild(delete_bam_files['tumor_rna'])
```

Comment:
what about `fusions.addChild(delete_bam_files['tumor_rna'])` That might be why you're getting key errors.